src/YoloDev.Expecto.TestSdk/settings.fs

```diff
@@ -14,57 +19,155 @@ type RunSettings =
+      confNode
+      |> Option.bind (Xml.element elementName)
+      |> Option.bind Xml.value
+      |> Option.bind parser
+
+  let readValueString elementName confNode = readValueParse Option.ofObj elementName confNode
+  let readValueBool elementName confNode = readValueParse TryParse.bool elementName confNode
+
+  let (|Element|_|) (name : string) (node : XElement) =
+    if name.Equals (node.Name.LocalName, StringComparison.OrdinalIgnoreCase)
+    then Some node.Value
+    else None
+
+  let (|Bool|_|) str = TryParse.bool str
+  let (|Int|_|) str = TryParse.int32 str
+  let (|Float|_|) str = TryParse.float str
+  let (|String|_|) (str : string) = Option.ofObj str
+
+  let readExpectoConfig expectoConfig (confNode: Xml.Linq.XElement) =
+    let configMapper node = 
+      match node with
+      | Element "sequenced" (Bool true) -> Some CLIArguments.Sequenced
+      | Element "sequenced" (Bool false) -> Some CLIArguments.Parallel
+      | Element "parallel" (Bool true) -> Some CLIArguments.Parallel
+      | Element "parallel" (Bool false) -> Some CLIArguments.Sequenced
+      | Element "parallel-workers" (Int i) -> Some (CLIArguments.Parallel_Workers i)
+      | Element "stress" (Float f) -> Some (CLIArguments.Stress f)
+      | Element "stress-timeout" (Float f) -> Some (CLIArguments.Stress_Timeout f)
+      | Element "stress-memory-limit" (Float f) -> Some (CLIArguments.Stress_Memory_Limit f)
+      | Element "stress-memory-limit" (Float f) -> Some (CLIArguments.Stress_Memory_Limit f)
+      | Element "fail-on-focused-tests" (Bool true) -> Some CLIArguments.Fail_On_Focused_Tests
+      | Element "debug" (Bool true) -> Some CLIArguments.Debug
+      | Element "log-name" (String s) -> Some (CLIArguments.Log_Name s)
+      | Element "filter" (String s) -> Some (CLIArguments.Filter s)
+      | Element "fscheck-max-tests" (Int i) -> Some (CLIArguments.FsCheck_Max_Tests i)
+      | Element "fscheck-start-size" (Int i) -> Some (CLIArguments.FsCheck_Max_Tests i)
+      | Element "fscheck-end-size" (Int i) -> Some (CLIArguments.FsCheck_End_Size i)
+      | Element "allow-duplicate-name" (Bool true) -> Some CLIArguments.Allow_Duplicate_Names
+      | Element "no-spinner" (Bool true) -> Some CLIArguments.No_Spinner
+      | Element "verbosity" (String s) -> Expecto.Logging.LogLevel.ofString s |> CLIArguments.Verbosity |> Some
+      | _ -> None
+
+    let args =
+      confNode.Descendants()
+      |> Seq.choose configMapper
+
+    (expectoConfig, args)
+    ||> Seq.fold(fun state next -> foldCLIArgumentToConfig next state)
     
-    let confNode = 
+
+  let read (runSettings: IRunSettings) =
+    let settings = defaultSettings
+
+    let runSettingsNode =
       Option.ofObj runSettings
       |> Option.bind (fun s -> Option.ofObj s.SettingsXml)
       |> Option.bind Xml.read
-      |> Option.bind Xml.root
-      |> Option.bind (Xml.element "RunSettings")
+      |> Option.bind Xml.root //This gets the RunSettings element
```

Comment:
This comment probably isn't needed. If this was wrong in advance, that just means I've confused myself, and this comment is just going to do the same 😛

Reply:
So I think this was broken before.  `Option.bind (Xml.element "RunSettings")` wouldn't find anything since  `Option.bind Xml.root` was the `RunSettings` element.  I just put it here for clarification.  I can remove it .